### ISSUE TITLE
feature request: atomic properties

### ISSUE THREAD

**Issue by zingale** [astropy/astropy#4403]
A lot of astrophysics deals with atomic or nuclear properties.  Is there any desire to add atomic properties (starting with, for example, element names, atmoic numbers and masses) to astropy?  This would help with computing mean molecular weights and molar fractions, which come up all the time in astro.

**Comment by pllim** [comment]
You mean, like https://pypi.python.org/pypi/periodictable ?

**Comment by eteq** [comment]
Hmm... I'm not sure we want to have a curated list of all this stuff in astropy itself, but I wonder if we want to consider packaging something like what @pllim pointed to as an `extern` package?  The main reason we might want to add an extra interface is so that it is compatible with `Quantity`/`astropy.units`. it looks like it probably wouldn't be too much work to add a layer over that package adding that...

**Comment by pllim** [comment]
Maybe start as an affiliated package that is a wrapper around it? What if its development is stopped? What if its API changes?

**Comment by zingale** [comment]
I'll poke around periodictable, which might be something nice to build upon.

On Tue, Dec 15, 2015 at 3:30 PM, P. L. Lim <notifications@github.com> wrote:

> Maybe start as an affiliated package that is a wrapper around it? What if
> its development is stopped? What if its API changes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/astropy/astropy/issues/4403#issuecomment-164887814>.
>



-- 
Michael Zingale
Associate Professor

Dept. of Physics & Astronomy • Stony Brook University • Stony Brook, NY
11794-3800
*phone*:  631-632-8225
*e-mail*: Michael.Zingale@stonybrook.edu
*web*: http://www.astro.sunysb.edu/mzingale
github: http://github.com/zingale

**Comment by crawfordsm** [comment]
Also check out https://github.com/astropy/astroquery as that has access to atomic databases via querying NIST as well as other libraries

**Comment by pllim** [comment]
Thanks, @crawfordsm . I see some promising ones under http://astroquery.readthedocs.org/en/latest/#other but I can't tell if they have all the requested info (like molecular weights) from the examples alone. Worth checking these out, @zingale .

**Comment by embray** [comment]
I could see a good argument for having this.  Agreed with the comments of @pllim and @eteq that it would be a significant new effort and might be best not tackling from scratch, but at the same time should be done in a way that interfaces nicely with the existing Astropy units/constants.  I'd be hesitant at this very early stage to make any prognostications about level of effort since we don't really know exactly what we're talking about here.  But I could see a place for this in Astropy--the feature would just need a champion, and probably an APE (remember those?)

**Comment by pllim** [comment]
For astronomy, we only really need `H_alpha`, right? Right? :wink:

**Comment by namurphy** [comment]
I just happened across this old issue.  We now have an [`atomic`](http://docs.plasmapy.org/en/latest/atomic/) subpackage in PlasmaPy that pretty much addresses this issue.  We should be interoperable with Astropy since we pretty heavily use `astropy.units`.  The most important feature is the [`Particle`](http://docs.plasmapy.org/en/latest/atomic/particle_class.html) class.  We can create a `Particle` instance for any ion and most fundamental particles.  The instance provides access to things like mass, charge, etc.

The story behind this subpackage is that when I started to write a function for the Alfvén speed, I realized that we needed a way to access the masses of arbitrary ions.  Several months later, after getting most of `atomic` written, we finally had a clean way to calculate the Alfvén speed!
